In [1]:
import agate

In [2]:
registered_voters = agate.Table.from_csv('Data/registeredvoters.csv')

In [3]:
print(registered_voters)

| column        | data_type |
| ------------- | --------- |
| County        | Text      |
| Republican10  | Number    |
| Democrat10    | Number    |
| Libertarian10 | Number    |
| Nonpartisan10 | Number    |
| Total10       | Number    |
| Republican16  | Number    |
| Democrat16    | Number    |
| Nonpartisan16 | Number    |
| Libertarian16 | Number    |
| Total16       | Number    |



In [4]:
registered_voters.print_table()

| County    | Republican10 | Democrat10 | Libertarian10 | Nonpartisan10 | Total10 | ... |
| --------- | ------------ | ---------- | ------------- | ------------- | ------- | --- |
| Adams     |       10,018 |      5,536 |             6 |         2,972 |  18,532 | ... |
| Antelope  |        3,005 |      1,147 |             0 |           538 |   4,690 | ... |
| Arthur    |          284 |         52 |             0 |            10 |     346 | ... |
| Banner    |          424 |         53 |             0 |            53 |     530 | ... |
| Blaine    |          314 |         56 |             0 |            24 |     394 | ... |
| Boone     |        2,390 |      1,156 |             0 |           408 |   3,954 | ... |
| Box Butte |        4,115 |      2,347 |             3 |         1,286 |   7,751 | ... |
| Boyd      |        1,036 |        338 |             0 |           129 |   1,503 | ... |
| Brown     |        1,663 |        363 |             0 |           224 |   2,250 | ... |
| Buffalo 

**SELECTED FIELDS: COUNTY, TOTAL VOTERS 2010 AND TOTAL VOTERS 2016**

In [5]:
for_printing = registered_voters.select(['County', 'Total10', 'Total16'])

In [6]:
for_printing.print_table()

| County    | Total10 | Total16 |
| --------- | ------- | ------- |
| Adams     |  18,532 |  19,527 |
| Antelope  |   4,690 |   4,557 |
| Arthur    |     346 |     341 |
| Banner    |     530 |     545 |
| Blaine    |     394 |     384 |
| Boone     |   3,954 |   3,785 |
| Box Butte |   7,751 |   7,577 |
| Boyd      |   1,503 |   1,494 |
| Brown     |   2,250 |   2,134 |
| Buffalo   |  27,104 |  29,193 |
| Burt      |   5,049 |   4,893 |
| Butler    |   6,117 |   5,585 |
| Cass      |  17,412 |  17,310 |
| Cedar     |   6,189 |   5,960 |
| Chase     |   2,616 |   2,486 |
| Cherry    |   4,158 |   4,185 |
| Cheyenne  |   6,900 |   6,863 |
| Clay      |   4,599 |   4,258 |
| Colfax    |   5,345 |   5,069 |
| Cuming    |   5,879 |   5,787 |
| ...       |     ... |     ... |


In [50]:
counties_rank = for_printing.order_by('Total16', reverse=True)

In [51]:
counties_rank.print_table()

| County       | Total10 | Total16 |
| ------------ | ------- | ------- |
| Douglas      | 313,266 | 345,966 |
| Lancaster    | 177,813 | 187,050 |
| Sarpy        |  92,785 | 108,473 |
| Hall         |  31,674 |  33,026 |
| Buffalo      |  27,104 |  29,193 |
| Scotts Bluff |  23,871 |  24,139 |
| Lincoln      |  23,213 |  24,012 |
| Dodge        |  21,876 |  22,113 |
| Madison      |  21,372 |  21,462 |
| Platte       |  19,435 |  20,116 |
| Adams        |  18,532 |  19,527 |
| Cass         |  17,412 |  17,310 |
| Saunders     |  14,181 |  14,696 |
| Washington   |  13,934 |  14,322 |
| Gage         |  15,202 |  14,136 |
| Dawson       |  13,578 |  13,916 |
| Seward       |  10,902 |  10,926 |
| Dakota       |  10,088 |  10,700 |
| Otoe         |  10,771 |  10,447 |
| York         |  10,790 |  10,013 |
| ...          |     ... |     ... |


**TOP TEN COUNTIES FOR REGISTERED VOTERS IN 2016: DOUGLAS, LANCASTER, SARPY, HALL, BUFFALO, SCOTTS BLUFF, LINCOLN, DODGE, MADISON, AND PLATTE**

***TOTAL VOTERS PERCENT CHANGE 2010-2016***

In [7]:
change = for_printing.compute([
    ('change', agate.PercentChange('Total10', 'Total16')),
])

In [8]:
change.print_table()

| County    | Total10 | Total16 |  change |
| --------- | ------- | ------- | ------- |
| Adams     |  18,532 |  19,527 |  5.369… |
| Antelope  |   4,690 |   4,557 | -2.836… |
| Arthur    |     346 |     341 | -1.445… |
| Banner    |     530 |     545 |  2.830… |
| Blaine    |     394 |     384 | -2.538… |
| Boone     |   3,954 |   3,785 | -4.274… |
| Box Butte |   7,751 |   7,577 | -2.245… |
| Boyd      |   1,503 |   1,494 | -0.599… |
| Brown     |   2,250 |   2,134 | -5.156… |
| Buffalo   |  27,104 |  29,193 |  7.707… |
| Burt      |   5,049 |   4,893 | -3.090… |
| Butler    |   6,117 |   5,585 | -8.697… |
| Cass      |  17,412 |  17,310 | -0.586… |
| Cedar     |   6,189 |   5,960 | -3.700… |
| Chase     |   2,616 |   2,486 | -4.969… |
| Cherry    |   4,158 |   4,185 |  0.649… |
| Cheyenne  |   6,900 |   6,863 | -0.536… |
| Clay      |   4,599 |   4,258 | -7.415… |
| Colfax    |   5,345 |   5,069 | -5.164… |
| Cuming    |   5,879 |   5,787 | -1.565… |
| ...       |     ... |     ... 

**TOP TEN FASTEST SHRINKING COUNTIES IN TERMS OF TOTAL REGISTERED VOTERS 2010-2016**

In [9]:
sorted_counties = change.order_by('change')

In [10]:
sorted_counties.print_table(max_rows=10)

| County     | Total10 | Total16 |   change |
| ---------- | ------- | ------- | -------- |
| Greeley    |   1,817 |   1,594 | -12.273… |
| Hooker     |     609 |     542 | -11.002… |
| Nemaha     |   5,011 |   4,563 |  -8.940… |
| Butler     |   6,117 |   5,585 |  -8.697… |
| Richardson |   6,187 |   5,669 |  -8.372… |
| Webster    |   2,613 |   2,413 |  -7.654… |
| Clay       |   4,599 |   4,258 |  -7.415… |
| York       |  10,790 |  10,013 |  -7.201… |
| Wheeler    |     654 |     607 |  -7.187… |
| Dundy      |   1,368 |   1,272 |  -7.018… |
| ...        |     ... |     ... |      ... |


In [11]:
from decimal import Decimal

def round_change(row):
    return row['change'].quantize(Decimal('0.1'))

rounded_change = sorted_counties.compute([
    ('change_rounded', agate.Formula(agate.Number(), round_change))
])

In [12]:
rounded_change.print_table()

| County     | Total10 | Total16 |   change | change_rounded |
| ---------- | ------- | ------- | -------- | -------------- |
| Greeley    |   1,817 |   1,594 | -12.273… |          -12.3 |
| Hooker     |     609 |     542 | -11.002… |          -11.0 |
| Nemaha     |   5,011 |   4,563 |  -8.940… |           -8.9 |
| Butler     |   6,117 |   5,585 |  -8.697… |           -8.7 |
| Richardson |   6,187 |   5,669 |  -8.372… |           -8.4 |
| Webster    |   2,613 |   2,413 |  -7.654… |           -7.7 |
| Clay       |   4,599 |   4,258 |  -7.415… |           -7.4 |
| York       |  10,790 |  10,013 |  -7.201… |           -7.2 |
| Wheeler    |     654 |     607 |  -7.187… |           -7.2 |
| Dundy      |   1,368 |   1,272 |  -7.018… |           -7.0 |
| Gage       |  15,202 |  14,136 |  -7.012… |           -7.0 |
| Keya Paha  |     686 |     639 |  -6.851… |           -6.9 |
| Hayes      |     769 |     720 |  -6.372… |           -6.4 |
| Jefferson  |   5,318 |   4,981 |  -6.337… |          

In [13]:
sorted_counties = rounded_change.order_by('change')

In [14]:
sorted_counties.print_table(max_rows=10)

| County     | Total10 | Total16 |   change | change_rounded |
| ---------- | ------- | ------- | -------- | -------------- |
| Greeley    |   1,817 |   1,594 | -12.273… |          -12.3 |
| Hooker     |     609 |     542 | -11.002… |          -11.0 |
| Nemaha     |   5,011 |   4,563 |  -8.940… |           -8.9 |
| Butler     |   6,117 |   5,585 |  -8.697… |           -8.7 |
| Richardson |   6,187 |   5,669 |  -8.372… |           -8.4 |
| Webster    |   2,613 |   2,413 |  -7.654… |           -7.7 |
| Clay       |   4,599 |   4,258 |  -7.415… |           -7.4 |
| York       |  10,790 |  10,013 |  -7.201… |           -7.2 |
| Wheeler    |     654 |     607 |  -7.187… |           -7.2 |
| Dundy      |   1,368 |   1,272 |  -7.018… |           -7.0 |
| ...        |     ... |     ... |      ... |            ... |


**TOP TEN FASTEST SHRINKING TOTAL REGISTERED VOTERS COUNTIES 2010-2016: GREELEY (-12.3%), HOOKER (-11%), NEMAHA (-8.9%), BUTLER (-8.7%), RICHARDSON (-8.4%), WEBSTER (-7.7%), CLAY (-7.4%), YORK(-7.2%), WHEELER (-7.2%), AND DUNDY (-7.0%) shrinking counties can be written, a newsworthy to write 12.3 pct shrinking**

In [15]:
sorted_counties = rounded_change.order_by('change', reverse=True)

In [16]:
sorted_counties.print_table(max_rows=10)

| County    | Total10 | Total16 |  change | change_rounded |
| --------- | ------- | ------- | ------- | -------------- |
| Sarpy     |  92,785 | 108,473 | 16.908… |           16.9 |
| Douglas   | 313,266 | 345,966 | 10.438… |           10.4 |
| Buffalo   |  27,104 |  29,193 |  7.707… |            7.7 |
| Dakota    |  10,088 |  10,700 |  6.067… |            6.1 |
| Adams     |  18,532 |  19,527 |  5.369… |            5.4 |
| Lancaster | 177,813 | 187,050 |  5.195… |            5.2 |
| Hall      |  31,674 |  33,026 |  4.268… |            4.3 |
| Saunders  |  14,181 |  14,696 |  3.632… |            3.6 |
| Platte    |  19,435 |  20,116 |  3.504… |            3.5 |
| Lincoln   |  23,213 |  24,012 |  3.442… |            3.4 |
| ...       |     ... |     ... |     ... |            ... |


**TOP TEN FASTEST GROWING OF THE TOTAL REGISTERED VOTERS COUNTIES 2010-2016: SARPY (16.9%), DOUGLAS (10.4%), BUFFALO (7.7%), DAKOTA (6.1%), ADAMS (5.4%), LANCASTER (5.2%), HALL (4.3%), SAUNDERS(3.6%), PLATTE (3.5%), AND LINCOLN (3.4%)**

***PART TWO DEMOCRAT, REPUBLICAN AND NONPARTISAN PCT CHANGE**

*CALCULATIONS*

In [18]:
registered_voters.print_table()

| County    | Republican10 | Democrat10 | Libertarian10 | Nonpartisan10 | Total10 | ... |
| --------- | ------------ | ---------- | ------------- | ------------- | ------- | --- |
| Adams     |       10,018 |      5,536 |             6 |         2,972 |  18,532 | ... |
| Antelope  |        3,005 |      1,147 |             0 |           538 |   4,690 | ... |
| Arthur    |          284 |         52 |             0 |            10 |     346 | ... |
| Banner    |          424 |         53 |             0 |            53 |     530 | ... |
| Blaine    |          314 |         56 |             0 |            24 |     394 | ... |
| Boone     |        2,390 |      1,156 |             0 |           408 |   3,954 | ... |
| Box Butte |        4,115 |      2,347 |             3 |         1,286 |   7,751 | ... |
| Boyd      |        1,036 |        338 |             0 |           129 |   1,503 | ... |
| Brown     |        1,663 |        363 |             0 |           224 |   2,250 | ... |
| Buffalo 

In [19]:
for_printing_2 = registered_voters.select(['County', 'Republican10', 'Republican16', 'Democrat10', 'Democrat16', 'Libertarian10', 'Libertarian16'])

In [20]:
for_printing_2.print_table()

| County    | Republican10 | Republican16 | Democrat10 | Democrat16 | Libertarian10 | ... |
| --------- | ------------ | ------------ | ---------- | ---------- | ------------- | --- |
| Adams     |       10,018 |       10,746 |      5,536 |      5,027 |             6 | ... |
| Antelope  |        3,005 |        3,088 |      1,147 |        863 |             0 | ... |
| Arthur    |          284 |          286 |         52 |         37 |             0 | ... |
| Banner    |          424 |          427 |         53 |         38 |             0 | ... |
| Blaine    |          314 |          310 |         56 |         43 |             0 | ... |
| Boone     |        2,390 |        2,469 |      1,156 |        901 |             0 | ... |
| Box Butte |        4,115 |        4,278 |      2,347 |      1,852 |             3 | ... |
| Boyd      |        1,036 |        1,084 |        338 |        250 |             0 | ... |
| Brown     |        1,663 |        1,658 |        363 |        253 |           

**I. REPUBLICAN PERCENT CHANGE**

In [21]:
for_printing_Republican = registered_voters.select(['County', 'Republican10', 'Republican16'])

In [22]:
for_printing_Republican. print_table()

| County    | Republican10 | Republican16 |
| --------- | ------------ | ------------ |
| Adams     |       10,018 |       10,746 |
| Antelope  |        3,005 |        3,088 |
| Arthur    |          284 |          286 |
| Banner    |          424 |          427 |
| Blaine    |          314 |          310 |
| Boone     |        2,390 |        2,469 |
| Box Butte |        4,115 |        4,278 |
| Boyd      |        1,036 |        1,084 |
| Brown     |        1,663 |        1,658 |
| Buffalo   |       15,768 |       16,974 |
| Burt      |        2,521 |        2,540 |
| Butler    |        3,044 |        3,066 |
| Cass      |        8,216 |        8,472 |
| Cedar     |        3,072 |        3,503 |
| Chase     |        1,827 |        1,830 |
| Cherry    |        2,945 |        3,154 |
| Cheyenne  |        4,146 |        4,250 |
| Clay      |        2,889 |        2,743 |
| Colfax    |        2,364 |        2,332 |
| Cuming    |        3,639 |        3,635 |
| ...       |          ... |    

In [23]:
change_Republican = for_printing_Republican.compute([
    ('change', agate.PercentChange('Republican10', 'Republican16')),
])

In [24]:
change_Republican.print_table()

| County    | Republican10 | Republican16 |  change |
| --------- | ------------ | ------------ | ------- |
| Adams     |       10,018 |       10,746 |  7.267… |
| Antelope  |        3,005 |        3,088 |  2.762… |
| Arthur    |          284 |          286 |  0.704… |
| Banner    |          424 |          427 |  0.708… |
| Blaine    |          314 |          310 | -1.274… |
| Boone     |        2,390 |        2,469 |  3.305… |
| Box Butte |        4,115 |        4,278 |  3.961… |
| Boyd      |        1,036 |        1,084 |  4.633… |
| Brown     |        1,663 |        1,658 | -0.301… |
| Buffalo   |       15,768 |       16,974 |  7.648… |
| Burt      |        2,521 |        2,540 |  0.754… |
| Butler    |        3,044 |        3,066 |  0.723… |
| Cass      |        8,216 |        8,472 |  3.116… |
| Cedar     |        3,072 |        3,503 | 14.030… |
| Chase     |        1,827 |        1,830 |  0.164… |
| Cherry    |        2,945 |        3,154 |  7.097… |
| Cheyenne  |        4,146 |

**REPUBLICAN TOP TEN COUNTIES OF THE MOST GROWING TO THE LEAST GROWING VOTERS**

In [25]:
sorted_counties_Republican = change_Republican.order_by('change', reverse=True)

In [26]:
sorted_counties_Republican.print_table()

| County     | Republican10 | Republican16 |  change |
| ---------- | ------------ | ------------ | ------- |
| Dixon      |        1,887 |        2,272 | 20.403… |
| Sarpy      |       43,657 |       50,057 | 14.660… |
| Cedar      |        3,072 |        3,503 | 14.030… |
| Saunders   |        7,229 |        8,124 | 12.381… |
| Grant      |          381 |          418 |  9.711… |
| Platte     |       10,998 |       12,051 |  9.574… |
| Lincoln    |       12,419 |       13,574 |  9.300… |
| Howard     |        2,241 |        2,448 |  9.237… |
| Greeley    |          578 |          626 |  8.304… |
| Washington |        7,613 |        8,201 |  7.724… |
| Buffalo    |       15,768 |       16,974 |  7.648… |
| Red Willow |        4,300 |        4,621 |  7.465… |
| Adams      |       10,018 |       10,746 |  7.267… |
| Douglas    |      120,973 |      129,688 |  7.204… |
| Cherry     |        2,945 |        3,154 |  7.097… |
| Hamilton   |        4,107 |        4,365 |  6.282… |
| Knox    

In [27]:
from decimal import Decimal

def round_change(row):
    return row['change'].quantize(Decimal('0.1'))

rounded_change = sorted_counties_Republican.compute([
    ('change_rounded', agate.Formula(agate.Number(), round_change))
])

In [28]:
rounded_change.print_table()

| County     | Republican10 | Republican16 |  change | change_rounded |
| ---------- | ------------ | ------------ | ------- | -------------- |
| Dixon      |        1,887 |        2,272 | 20.403… |           20.4 |
| Sarpy      |       43,657 |       50,057 | 14.660… |           14.7 |
| Cedar      |        3,072 |        3,503 | 14.030… |           14.0 |
| Saunders   |        7,229 |        8,124 | 12.381… |           12.4 |
| Grant      |          381 |          418 |  9.711… |            9.7 |
| Platte     |       10,998 |       12,051 |  9.574… |            9.6 |
| Lincoln    |       12,419 |       13,574 |  9.300… |            9.3 |
| Howard     |        2,241 |        2,448 |  9.237… |            9.2 |
| Greeley    |          578 |          626 |  8.304… |            8.3 |
| Washington |        7,613 |        8,201 |  7.724… |            7.7 |
| Buffalo    |       15,768 |       16,974 |  7.648… |            7.6 |
| Red Willow |        4,300 |        4,621 |  7.465… |          

**TOP TEN COUNTIES THAT GROWING FASTEST FROM 2010 TO 2016: DIXON (20.4%), SARPY (14.7%), CEDAR (14%), SAUNDERS(12.4%), GRANT (9.7%), PLATTE(9.6%), LINCOLN(9.3%), HOWARD(9,.2%), GREELEY(8.3%), AND WASHINGTON (7.7%)**

In [29]:
rounded_change.print_table(max_rows=10)

| County     | Republican10 | Republican16 |  change | change_rounded |
| ---------- | ------------ | ------------ | ------- | -------------- |
| Dixon      |        1,887 |        2,272 | 20.403… |           20.4 |
| Sarpy      |       43,657 |       50,057 | 14.660… |           14.7 |
| Cedar      |        3,072 |        3,503 | 14.030… |           14.0 |
| Saunders   |        7,229 |        8,124 | 12.381… |           12.4 |
| Grant      |          381 |          418 |  9.711… |            9.7 |
| Platte     |       10,998 |       12,051 |  9.574… |            9.6 |
| Lincoln    |       12,419 |       13,574 |  9.300… |            9.3 |
| Howard     |        2,241 |        2,448 |  9.237… |            9.2 |
| Greeley    |          578 |          626 |  8.304… |            8.3 |
| Washington |        7,613 |        8,201 |  7.724… |            7.7 |
| ...        |          ... |          ... |     ... |            ... |


**II. DEMOCRAT PERCENT CHANGE**

In [30]:
for_printing_Democrat = registered_voters.select(['County', 'Democrat10', 'Democrat16'])

In [31]:
for_printing_Democrat.print_table()

| County    | Democrat10 | Democrat16 |
| --------- | ---------- | ---------- |
| Adams     |      5,536 |      5,027 |
| Antelope  |      1,147 |        863 |
| Arthur    |         52 |         37 |
| Banner    |         53 |         38 |
| Blaine    |         56 |         43 |
| Boone     |      1,156 |        901 |
| Box Butte |      2,347 |      1,852 |
| Boyd      |        338 |        250 |
| Brown     |        363 |        253 |
| Buffalo   |      6,785 |      6,453 |
| Burt      |      1,694 |      1,440 |
| Butler    |      2,134 |      1,587 |
| Cass      |      5,435 |      4,633 |
| Cedar     |      2,231 |      1,574 |
| Chase     |        526 |        371 |
| Cherry    |        780 |        566 |
| Cheyenne  |      1,484 |      1,221 |
| Clay      |      1,081 |        848 |
| Colfax    |      2,394 |      2,018 |
| Cuming    |      1,485 |      1,240 |
| ...       |        ... |        ... |


In [32]:
change_Democrat = for_printing_Democrat.compute([
    ('change', agate.PercentChange('Democrat10', 'Democrat16')),
])

In [33]:
change_Democrat.print_table()

| County    | Democrat10 | Democrat16 |   change |
| --------- | ---------- | ---------- | -------- |
| Adams     |      5,536 |      5,027 |  -9.194… |
| Antelope  |      1,147 |        863 | -24.760… |
| Arthur    |         52 |         37 | -28.846… |
| Banner    |         53 |         38 | -28.302… |
| Blaine    |         56 |         43 | -23.214… |
| Boone     |      1,156 |        901 | -22.059… |
| Box Butte |      2,347 |      1,852 | -21.091… |
| Boyd      |        338 |        250 | -26.036… |
| Brown     |        363 |        253 | -30.303… |
| Buffalo   |      6,785 |      6,453 |  -4.893… |
| Burt      |      1,694 |      1,440 | -14.994… |
| Butler    |      2,134 |      1,587 | -25.633… |
| Cass      |      5,435 |      4,633 | -14.756… |
| Cedar     |      2,231 |      1,574 | -29.449… |
| Chase     |        526 |        371 | -29.468… |
| Cherry    |        780 |        566 | -27.436… |
| Cheyenne  |      1,484 |      1,221 | -17.722… |
| Clay      |      1,081 |     

In [34]:
sorted_counties_Democrat = change_Democrat.order_by('change', reverse=True)

In [35]:
sorted_counties_Democrat.print_table()

| County       | Democrat10 | Democrat16 |   change |
| ------------ | ---------- | ---------- | -------- |
| Sarpy        |     27,188 |     29,538 |   8.644… |
| Douglas      |    125,194 |    132,466 |   5.809… |
| Thurston     |      2,514 |      2,582 |   2.705… |
| Lancaster    |     67,217 |     68,127 |   1.354… |
| Dawson       |      3,620 |      3,660 |   1.105… |
| Dakota       |      4,238 |      4,255 |   0.401… |
| Hall         |      9,824 |      9,538 |  -2.911… |
| Buffalo      |      6,785 |      6,453 |  -4.893… |
| Loup         |         83 |         77 |  -7.229… |
| Dawes        |      1,422 |      1,316 |  -7.454… |
| Wayne        |      1,204 |      1,101 |  -8.555… |
| Adams        |      5,536 |      5,027 |  -9.194… |
| Stanton      |        832 |        752 |  -9.615… |
| Madison      |      5,425 |      4,888 |  -9.899… |
| Saline       |      3,709 |      3,303 | -10.946… |
| Scotts Bluff |      6,821 |      6,067 | -11.054… |
| Dodge        |      7,132 

In [37]:
from decimal import Decimal

def round_change(row):
    return row['change'].quantize(Decimal('0.1'))

rounded_change = sorted_counties_Democrat.compute([
    ('change_rounded', agate.Formula(agate.Number(), round_change))
])

In [38]:
rounded_change.print_table()

| County       | Democrat10 | Democrat16 |   change | change_rounded |
| ------------ | ---------- | ---------- | -------- | -------------- |
| Sarpy        |     27,188 |     29,538 |   8.644… |            8.6 |
| Douglas      |    125,194 |    132,466 |   5.809… |            5.8 |
| Thurston     |      2,514 |      2,582 |   2.705… |            2.7 |
| Lancaster    |     67,217 |     68,127 |   1.354… |            1.4 |
| Dawson       |      3,620 |      3,660 |   1.105… |            1.1 |
| Dakota       |      4,238 |      4,255 |   0.401… |            0.4 |
| Hall         |      9,824 |      9,538 |  -2.911… |           -2.9 |
| Buffalo      |      6,785 |      6,453 |  -4.893… |           -4.9 |
| Loup         |         83 |         77 |  -7.229… |           -7.2 |
| Dawes        |      1,422 |      1,316 |  -7.454… |           -7.5 |
| Wayne        |      1,204 |      1,101 |  -8.555… |           -8.6 |
| Adams        |      5,536 |      5,027 |  -9.194… |           -9.2 |
| Stan

**TOP TEN COUNTIES THAT GROWING FASTEST FROM 2010 TO 2016: SARPY (8.6%), DOUGLAS (5.8%), THURSTON (2.7%), LANCASTER (1.4%), DAWSON(1.1%), DAKOTA(0.4%), HALL (-2.9%), BUFFALO (-4.9%), LOUP(-7.2%), AND DAWES (-7.5%)**

In [39]:
rounded_change.print_table(max_rows=10)

| County    | Democrat10 | Democrat16 |  change | change_rounded |
| --------- | ---------- | ---------- | ------- | -------------- |
| Sarpy     |     27,188 |     29,538 |  8.644… |            8.6 |
| Douglas   |    125,194 |    132,466 |  5.809… |            5.8 |
| Thurston  |      2,514 |      2,582 |  2.705… |            2.7 |
| Lancaster |     67,217 |     68,127 |  1.354… |            1.4 |
| Dawson    |      3,620 |      3,660 |  1.105… |            1.1 |
| Dakota    |      4,238 |      4,255 |  0.401… |            0.4 |
| Hall      |      9,824 |      9,538 | -2.911… |           -2.9 |
| Buffalo   |      6,785 |      6,453 | -4.893… |           -4.9 |
| Loup      |         83 |         77 | -7.229… |           -7.2 |
| Dawes     |      1,422 |      1,316 | -7.454… |           -7.5 |
| ...       |        ... |        ... |     ... |            ... |


**III. INDEPENDENTS (Nonpartisan) PERCENT CHANGE**

In [41]:
for_printing_Nonpartisan = registered_voters.select(['County', 'Nonpartisan10', 'Nonpartisan16'])

In [42]:
for_printing_Nonpartisan.print_table()

| County    | Nonpartisan10 | Nonpartisan16 |
| --------- | ------------- | ------------- |
| Adams     |         2,972 |         3,591 |
| Antelope  |           538 |           594 |
| Arthur    |            10 |            15 |
| Banner    |            53 |            73 |
| Blaine    |            24 |            29 |
| Boone     |           408 |           404 |
| Box Butte |         1,286 |         1,395 |
| Boyd      |           129 |           156 |
| Brown     |           224 |           214 |
| Buffalo   |         4,537 |         5,461 |
| Burt      |           834 |           878 |
| Butler    |           939 |           913 |
| Cass      |         3,755 |         4,025 |
| Cedar     |           886 |           869 |
| Chase     |           263 |           277 |
| Cherry    |           433 |           455 |
| Cheyenne  |         1,266 |         1,336 |
| Clay      |           629 |           650 |
| Colfax    |           587 |           701 |
| Cuming    |           754 |     

In [43]:
change_Nonpartisan = for_printing_Nonpartisan.compute([
    ('change', agate.PercentChange('Nonpartisan10', 'Nonpartisan16')),
])

In [44]:
change_Nonpartisan.print_table()

| County    | Nonpartisan10 | Nonpartisan16 |  change |
| --------- | ------------- | ------------- | ------- |
| Adams     |         2,972 |         3,591 | 20.828… |
| Antelope  |           538 |           594 | 10.409… |
| Arthur    |            10 |            15 | 50.000… |
| Banner    |            53 |            73 | 37.736… |
| Blaine    |            24 |            29 | 20.833… |
| Boone     |           408 |           404 | -0.980… |
| Box Butte |         1,286 |         1,395 |  8.476… |
| Boyd      |           129 |           156 | 20.930… |
| Brown     |           224 |           214 | -4.464… |
| Buffalo   |         4,537 |         5,461 | 20.366… |
| Burt      |           834 |           878 |  5.276… |
| Butler    |           939 |           913 | -2.769… |
| Cass      |         3,755 |         4,025 |  7.190… |
| Cedar     |           886 |           869 | -1.919… |
| Chase     |           263 |           277 |  5.323… |
| Cherry    |           433 |           455 |  5

In [45]:
sorted_counties_Nonpartisan = change_Nonpartisan.order_by('change', reverse=True)

In [46]:
sorted_counties_Nonpartisan.print_table()

| County    | Nonpartisan10 | Nonpartisan16 |  change |
| --------- | ------------- | ------------- | ------- |
| McPherson |            15 |            24 | 60.000… |
| Arthur    |            10 |            15 | 50.000… |
| Thomas    |            34 |            49 | 44.118… |
| Keya Paha |            31 |            44 | 41.935… |
| Banner    |            53 |            73 | 37.736… |
| Sarpy     |        21,900 |        27,620 | 26.119… |
| Logan     |            54 |            68 | 25.926… |
| Thayer    |           480 |           594 | 23.750… |
| Kimball   |           451 |           552 | 22.395… |
| Boyd      |           129 |           156 | 20.930… |
| Blaine    |            24 |            29 | 20.833… |
| Adams     |         2,972 |         3,591 | 20.828… |
| Buffalo   |         4,537 |         5,461 | 20.366… |
| Saunders  |         2,098 |         2,524 | 20.305… |
| Douglas   |        66,976 |        80,315 | 19.916… |
| Colfax    |           587 |           701 | 19

In [47]:
from decimal import Decimal

def round_change(row):
    return row['change'].quantize(Decimal('0.1'))

rounded_change = sorted_counties_Nonpartisan.compute([
    ('change_rounded', agate.Formula(agate.Number(), round_change))
])

In [48]:
rounded_change.print_table()

| County    | Nonpartisan10 | Nonpartisan16 |  change | change_rounded |
| --------- | ------------- | ------------- | ------- | -------------- |
| McPherson |            15 |            24 | 60.000… |           60.0 |
| Arthur    |            10 |            15 | 50.000… |           50.0 |
| Thomas    |            34 |            49 | 44.118… |           44.1 |
| Keya Paha |            31 |            44 | 41.935… |           41.9 |
| Banner    |            53 |            73 | 37.736… |           37.7 |
| Sarpy     |        21,900 |        27,620 | 26.119… |           26.1 |
| Logan     |            54 |            68 | 25.926… |           25.9 |
| Thayer    |           480 |           594 | 23.750… |           23.8 |
| Kimball   |           451 |           552 | 22.395… |           22.4 |
| Boyd      |           129 |           156 | 20.930… |           20.9 |
| Blaine    |            24 |            29 | 20.833… |           20.8 |
| Adams     |         2,972 |         3,591 | 20.82

**TOP TEN COUNTIES THAT GROWING FASTEST FROM 2010 TO 2016: MCPHERSON (60%), ARTHUR (50%), THOMAS (44.1%), KAYA PAHA (41.9%), BANNER (37.7%), SARPY (26.1%), LOGAN (25.9%), THAYER (23.8%), KIMBALL (22.4%), AND BOYD (20.9%)**

In [49]:
rounded_change.print_table(max_rows=10)

| County    | Nonpartisan10 | Nonpartisan16 |  change | change_rounded |
| --------- | ------------- | ------------- | ------- | -------------- |
| McPherson |            15 |            24 | 60.000… |           60.0 |
| Arthur    |            10 |            15 | 50.000… |           50.0 |
| Thomas    |            34 |            49 | 44.118… |           44.1 |
| Keya Paha |            31 |            44 | 41.935… |           41.9 |
| Banner    |            53 |            73 | 37.736… |           37.7 |
| Sarpy     |        21,900 |        27,620 | 26.119… |           26.1 |
| Logan     |            54 |            68 | 25.926… |           25.9 |
| Thayer    |           480 |           594 | 23.750… |           23.8 |
| Kimball   |           451 |           552 | 22.395… |           22.4 |
| Boyd      |           129 |           156 | 20.930… |           20.9 |
| ...       |           ... |           ... |     ... |            ... |


**Answers all in bolds**